src/nvim/ex_docmd.c

```diff
@@ -9411,3 +9411,36 @@ static void ex_terminal(exarg_T *eap)
     xfree(name);
   }
 }
+
+static void ex_plug(exarg_T *eap)
+{
+  char *path = NULL;
+  if (*eap->arg != NUL) {
+    if (os_isdir(eap->arg)) {  // full path: plug ~/.config/nvim/plugged/coolplugin
+      path = (char *)vim_strsave_fnameescape(eap->arg, FALSE);
+    } else {  // spec: plug neovim/neovimplugin
+      char_u *user_vim = vim_strsave(p_rtp);
+      strtok((char *)user_vim, ","); // split(&runtimepath, ',')[0]
+      char_u *basename = path_tail((char_u *)eap->arg); // 'neovim/neovimplugin' -> 'neovimplugin'
```

Comment:
> Fair enough. Likewise, :plug should be able to work without there being a more featureful plugin manager. But a plugin manager should be able to use :plug if it wants to.

This is why I think that early+event is better. Current implementation is too limited and the steps hardcoded in C cannot be avoided.

I have though an alternative: first `plug` invocation should call `plugin#Initialize()`, and `autoload/plugin.vim` should then setup the events. To override default plugin other plugin managers should require user to prepend their path to `&runtimepath` before calling `:plug` or instell themselves to `~/.nvim` which is fine since they *already* require this (except that both appending and prepending is fine), in addition to calling initialization function by hand.